### PR TITLE
Initiate node sizing docs

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Add a doc entry to help users dimension their QW clusters.

### How was this PR tested?

Describe how you tested this PR.

### CODE REVIEW

File: docs/get-started/node-sizing.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+title: Node sizing
```
Comment (guilload) [inline docs/get-started/node-sizing.md:2]:
```suggestion
title: Cluster sizing
```

**Review by guilload — COMMENTED** [review]
I suggest we move this guide to the deployment section.

File: docs/get-started/node-sizing.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+[Kafka](../configuration/source-config.md#kafka-source).
+
+<!-- TODO: change this note when releasing ingest v2 -->
+
+:::
+
+
+### Searchers
+
+Search performance is highly dependent on the workload. For example term queries
```
Comment (guilload) [inline docs/get-started/node-sizing.md:44]:
```suggestion
Search performance is highly dependent on the workload. For example, term queries
```

**Review by guilload — APPROVED** [review]
Looking good.

I'd add a warning "you mileage may vary blabla, use this guide to establish baseline and increase or decrease the resources depending on your workload."

I would remove the file-backed section and create a dedicated metastore section with two subsections one for the service and one for the database. Up to you.

File: docs/get-started/node-sizing.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+## Quickwit services
+
+### Indexers
+
+Here are some high level guidelines to size your Indexer nodes:
+- Quickwit can index at around **7.5MB per second per core**
+- For the general usecase, configure 4GB of RAM per core
+  - Workloads with a large number of indexes or data sources consume more RAM
+    <!-- TODO: Mention cooperative indexing here ? -->
```
[the remark this answers is not in the document]
Comment (rdettai) [reply docs/get-started/node-sizing.md:21]: cool!

File: docs/get-started/node-sizing.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  - Don't use instances with less than 8GB of RAM
+    <!-- Note: 4GB for the heap size (per pipeline), and 2GB for ingest queues -->
```
[the remark this answers is not in the document]
Comment (rdettai) [reply docs/get-started/node-sizing.md:23]: > We also cap the number of indexing writers open concurrently.

Does this cap happen even without the cooperative indexing?

**Comment by rdettai** [comment]
> create a dedicated metastore section with two subsections one for the service and one for the database

In many setups, the Postgres config will be a bit separate from the QW cluster config (e.g they might have different lifecycles, so live in different terraform scripts). So I think its more readable for the typical ops user to have the sizing suggestions organized accordingly.

**Pull request merged**